Source/Core/Core/Core.cpp

```diff
@@ -855,6 +863,33 @@ void Callback_FramePresented(double actual_emulation_speed)
 // Called from VideoInterface::Update (CPU thread) at emulated field boundaries
 void Callback_NewField()
 {
+  if (!was_rng_seed_called)
+  {
+    LuaGameCubeButtonProbabilityEvent::SetRngSeeding();
+    was_rng_seed_called = true;
+  }
+
+  if (Lua::is_lua_script_active && !Lua::LuaEmu::waiting_for_save_state_load &&
+      !Lua::LuaEmu::waiting_for_save_state_save && !Lua::LuaEmu::waiting_to_start_playing_movie &&
+      !Lua::LuaEmu::waiting_to_save_movie)
+  {
+    for (int i = 0; i < 4; ++i)
+    {
+      Lua::LuaGameCubeController::overwrite_controller_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::add_to_controller_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::do_random_input_events_at_specified_port[i] = false;
+      Lua::LuaGameCubeController::random_button_events[i].clear();
+      Lua::LuaGameCubeController::button_lists_for_add_to_controller_inputs[i].clear();
+      memset(&Lua::LuaGameCubeController::new_overwrite_controller_inputs[i], 0,
+             sizeof(Movie::ControllerState));
+      memset(&Lua::LuaGameCubeController::add_to_controller_inputs[i], 0,
+             sizeof(Movie::ControllerState));
+    }
+
+    if (Lua::LuaOnFrameStartCallback::frame_start_callback_is_registered)
+      Lua::LuaOnFrameStartCallback::RunCallback();
```

Comment:
So you call this "OnFrameStart", but this isn't really the start of the frame, and is pretty disconnected with when the rendering of graphics happens. 

It's related to TV scanout timings and is some point about half-way though the vertical blanking region (not really close to anything important)

Reply:
Still, since it's the first function that runs after the frame counter gets incremented, I think it makes for a good intuitive definition of OnFrameStart (and to be fair, its kind of ambiguous in Dolphin trying to define where a frame ends/starts - so this is good enough for me).